dart/lib/src/sentry.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:meta/meta.dart';
+
+import 'client.dart';
+import 'protocol.dart';
+import 'sentry_options.dart';
+import 'stack_trace.dart';
+
+/// Sentry SDK main entry point
+///
+class Sentry {
+  static SentryClient _client;
+
+  Sentry._();
+
+  static void init(SentryOptions options) {
```

Comment:
instead of taking `SentryOptions` directly, if possible let's do it with a callback configurator, see:

https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/Sentry.java#L104-L111

https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/Sentry.java#L487-L500

so we have the change to instantiate SentryOptions ourselves and set default values, so people just mutate what they really need afterwards